tests/manual-test-cases/Group19-ROBO/19-2-ROBO-Container-Limit.md

```diff
@@ -0,0 +1,50 @@
+Test 19-2 - ROBO - Container VM Limit
+=======
+
+# Purpose:
+To verify that the total container VM limit feature works as expected in a vSphere ROBO Advanced environment.
+
+# References:
+1. [vSphere Remote Office and Branch Office](http://www.vmware.com/products/vsphere/remote-office-branch-office.html)
+2. [Limit total allowed containerVMs per VCH](https://github.com/vmware/vic/issues/7273)
+3. [vic-machine inspect to report configured containerVM limit](https://github.com/vmware/vic/issues/7284)
+
+# Environment:
+This test requires access to VMware Nimbus cluster for dynamic ESXi and vCenter creation. This test should be executed in the following topologies and should have vSAN enabled.
+* 1 vCenter host with 3 clusters, where 1 cluster has 1 ESXi host and the other 2 clusters have 3 ESXi hosts each
+* 2 vCenter hosts connected with ELM, where each vCenter host has a cluster/host/datacenter topology that emulates a customer environment (exact topology TBD)
+
+See https://confluence.eng.vmware.com/display/CNA/VIC+ROBO for more details.
+
+# Test Steps:
+1. Deploy a ROBO Advanced vCenter testbed for both environments above
+2. Install the VIC appliance on vCenter with a container VM limit of y
+3. Use vic-machine inspect to verify the set container VM limit
+4. Visit the VCH Admin page and verify the container VM limit is displayed in the VCH Info section
+5. Create and run y containers with the VCH
+6. Attempt to create/run more than a total of y containers with the VCH
+7. Use vic-machine configure to increase the container VM limit (new limit = z)
+8. Use vic-machine inspect to verify the new container VM limit
+9. Visit the VCH Admin page and verify the container VM limit is displayed in the VCH Info section
+10. Create more containers and verify that up to a total of z containers can be created
+11. Use vic-machine configure to set the limit to lower than the current number of containers created/running
+12. Attempt to create/run more containers
+13. Delete/stop some containers so the current container VM count is lower than the set limit
+14. Attempt to create/run more containers until the set limit
+15. Delete the VCH
+
```

Comment:
suggest we also try:
1. running `run/delete` in series to ensure that we're only enforcing concurrently running limits.
2. creating more than y cVMs but not starting them (assuming the enforcement is for running cVMs for first drop)
3. starting and deleting containers at the same time so we stay close to the limit but are bouncing off it and ensure after we stop with the concurrent operations that we can still hit the limit and are not over it - this tests the concurrency of the bound tracking.

Reply:
Fixed - thanks for clarifying!